src/main/scala/Day06.scala

```diff
@@ -97,14 +97,20 @@ object Day06 extends App:
       acc + dist
     })
 
-  val amountRegionPoints = {
-    val distancePoints = for {
-      i <- 0 until dim
-      j <- 0 until dim
-      dist = distanceToPoints(i, j, coordinates)
-    } yield dist
-    distancePoints.count(_ < 10000)
-  }
+//  val amountRegionPoints = {
+//    val distancePoints = for {
+//      i <- 0 until dim
+//      j <- 0 until dim
+//      dist = distanceToPoints(i, j, coordinates)
+//    } yield dist
+//    distancePoints.count(_ < 10000)
+//  }
+  val amountRegionPoints = (0 until dim).foldLeft(0)((acc, i) => {
```

Comment:
yup - that's one way to get rid of mutable state idd; a nested `foldLeft` passing the accumulator to the inner.

alternatively you could abstract over `x` and `y` coordinates, eg.:

```
case class Pos(x: Int, y: Int)
```

and collaps the grid into one dimension, eg.:

```
type Grid = List[Pos]
```

this works bc the selective criteria, manhattan distance, is defined in terms of 'positions'